packages/cli/src/commands/promote/request-promote.ts

```diff
@@ -29,6 +32,18 @@ export default async function requestPromote({
     output: client.output,
   });
 
+  if (deployment.target !== 'production' && !yes) {
+    const question =
+      'This deployment does not target production, therefore promotion will not apply production environment variables. Are you sure you want to continue?';
+    const answer = await confirm(client, question, true);
+    if (!answer) {
+      output.error(
+        'User declined to promote a deployment that did not target production.'
+      );
+      return 1;
```

Comment:
```suggestion
    const answer = await confirm(client, question, false);
    if (!answer) {
      output.error('Canceled');
      return 0;
```

Looks like the pattern here is to default to `false` and then false will print Canceled with 0 exit code.

See other commands here:

https://github.com/vercel/vercel/blob/36db62a491aa3155ebb286f0e533bd8edfc93670/packages/cli/src/commands/env/rm.ts#L104-L116

https://github.com/vercel/vercel/blob/36db62a491aa3155ebb286f0e533bd8edfc93670/packages/cli/src/commands/env/pull.ts#L88-L97